features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/APIDefinition/swaggerUI/PatchedSwaggerUIReact.jsx

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import swaggerUIConstructor, { presets } from 'swagger-ui';
+
+/**
+ *
+ *
+ * @export
+ * @class SwaggerUI
+ * @extends {React.Component}
+ */
+export default class PatchedSwaggerUI extends React.Component {
+    constructor(props) {
+        super(props);
+        this.SwaggerUIComponent = null;
+        this.system = null;
+    }
+
+    /**
+     *
+     *
+     * @memberof SwaggerUI
+     */
+    componentDidMount() {
+        const ui = swaggerUIConstructor({
+            spec: this.props.spec,
+            url: this.props.url,
+            defaultModelsExpandDepth: this.props.defaultModelsExpandDepth,
+            presets: [presets.apis, ...this.props.presets],
+            requestInterceptor: this.requestInterceptor,
+            responseInterceptor: this.responseInterceptor,
+            onComplete: this.onComplete,
+            docExpansion: this.props.docExpansion,
+            supportedSubmitMethods: [],
+        });
+
+        this.system = ui;
+        this.SwaggerUIComponent = ui.getComponent('App', 'root');
+
+        this.forceUpdate();
```

Comment:
Why we need to do a force update here ?

Reply:
Otherwise, it will not render in the component is loaded for the first time. We have to edit the definition. 
With forceUpdate(), it renders properly.

Reply:
Can't we keep the definition in a state ?, I think we should avoid using forceUpdate here, Which change cause a difference in forceUpdate rendered output here?